File: sdk/agid-julia/src/AGID.jl
module AGID

export encode, decode, AGIDResult

const BASE32_ALPHABET = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
const PREFIX_LENGTH = 2
const HASH_LENGTH = 10
const TOTAL_LENGTH = 12

struct AGIDResult
    id::String
    lat::Float64
    lon::Float64
    face::Union{Int, Nothing}
end

function encode(lat::Real, lon::Real)
    error("wire this package to the AGID reference implementation")
end

decode(id::AbstractString) = nothing

end

File: sdk/agid-swift/Sources/AGID/AGID.swift
public let base32Alphabet = "0123456789ABCDEFGHJKMNPQRSTVWXYZ"
public let agidPrefixLength = 2
public let agidHashLength = 10
public let agidTotalLength = 12

public struct AGIDResult: Equatable {
    public let id: String
    public let lat: Double
    public let lon: Double
    public let face: Int?
}

public func encode(lat: Double, lon: Double) throws -> AGIDResult {
    throw AGIDError.notImplemented
}

public func decode(_ id: String) -> AGIDResult? {
    nil
}

public enum AGIDError: Error {
    case notImplemented
}
